File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueStatusChangedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Issue;

import java.time.Instant;

public record IssueStatusChangedEvent(
        long projectId,
        long issueId,
        long updaterId,
        String oldStatus,
        String newStatus,
        Instant changedAtUtc
) {
    public static IssueStatusChangedEvent from(Issue issue, long updaterId, String oldStatus) {
        return new IssueStatusChangedEvent(
                issue.getProjectId(),
                issue.getId(),
                updaterId,
                oldStatus,
                issue.getStatus().name(),
                Instant.now()
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueStatusChangedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueStatusChangedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("updaterId")]
    public long UpdaterId { get; set; }

    [JsonPropertyName("oldStatus")]
    public string OldStatus { get; set; } = string.Empty;

    [JsonPropertyName("newStatus")]
    public string NewStatus { get; set; } = string.Empty;

    [JsonPropertyName("changedAtUtc")]
    public DateTime ChangedAtUtc { get; set; } = DateTime.UtcNow;
}